### ISSUE TITLE
Ionic Build Error, cannot find GoogleService-Info.plist

### ISSUE THREAD

**Issue by jvjvjv** [phonegap/phonegap-plugin-push#1826]
Hi. I'm trying to Build, using 2.0.0-rc4 per https://github.com/phonegap/phonegap-plugin-push/issues/1767 and https://github.com/phonegap/phonegap-plugin-push/issues/1701 

### Expected Behaviour
Project builds

### Actual Behaviour
Project does not build, giving error 

        The following build commands failed:
        CopyPlistFile
        /Users/jason/Library/Developer/Xcode/DerivedData/MyProject-hcncjnpsinwglhghbzfnuivgzjeg/Build/Intermediates/ArchiveIntermediates/MyProject/InstallationBuildProductsLocation/Applications/MyProject.app/GoogleService-Info.plist
        /Users/jason/jason/code/GIT/MyProject/mobile/platforms/ios/MyProject/Resources/GoogleService-Info.plist
        (1 failure)

### Reproduce Scenario (including but not limited to)
* Initial build after adding plugin

#### Steps to Reproduce 
* Add plugin per [INSTALLATION.md](https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/INSTALLATION.md)
* Download and place ```google-services.json``` and ```GoogleService-Info.plist``` to ```/Users/jason/code/GIT/MyProject/mobile``` directory
* Run ```ionic cordova build ios```

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
n/a

#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
n/a

#### Cordova CLI version and cordova platform version

    ionic 3.4.0. (see above)

#### Plugin version

    phonegap-plugin-push 2.0.0-rc4
    phonegap-plugin-push 1.10.5 (may have provided different error during build)

#### Sample Push Data Payload
n/a
#### Sample Code that illustrates the problem
n/a

**Comment by macdonst** [comment]
@jvjvjv what version of cordova-ios is ionic using under the hood?

**Comment by jvjvjv** [comment]
Thanks for checking in. I had provided that in a previous draft of the above message but it looks like I'd removed it by accident.

    cordova-ios 4.4.0

**Comment by jvjvjv** [comment]
Here's a question, @macdonst. Why does the destination directory duplicate my name? The project is under ```/Users/jason/code/GIT/MyProject/mobile/```  and not ```/Users/jason/jason/code/GIT/MyProject/mobile/```.

**Comment by jvjvjv** [comment]
Here's some more information. It's not Ionic.

    cordova create bonsaitest com.bonsai.test "Boina"
    cd bonsaitest
    cordova plugin add phonegap-plugin-push@2.0.0-rc4
    cordova build ios

This also fails. Using Cordova@7.0.1, cordova-ios@4.4.0

**Comment by jvjvjv** [comment]
Here's some more information. It's not Ionic.

    cordova create bonsaitest com.bonsai.test "Boina"
    cd bonsaitest
    cordova plugin add phonegap-plugin-push@2.0.0-rc4
    nano config.xml (add GoogleService-Info.plist file)
    cordova build ios

This also fails. with the same type of error.

    The following build commands failed:
        CopyPlistFile /Users/Jason/Code/GIT/bonsaitest/platforms/ios/build/emulator/Boina.app/GoogleService-Info.plist /Users/jason/jason/code/GIT/bonsaitest/platforms/ios/Boina/Resources/GoogleService-Info.plist
(1 failure)

 Using Cordova@7.0.1, cordova-ios@4.4.0

**Comment by macdonst** [comment]
@jvjvjv I have no idea why this is happening but can you paste the line you add to config.xml?

**Comment by jvjvjv** [comment]
```<resource-file src="GoogleService-Info.plist" target="GoogleService-Info.plist" />```

**Comment by macdonst** [comment]
@jvjvjv try this (omit target):

```
<resource-file src="GoogleService-Info.plist" />
```

**Comment by jvjvjv** [comment]
I happened to have done that as well; it didn't work so based on https://github.com/phonegap/phonegap-plugin-push/issues/1701's comments I tried adding target.

**Comment by jvjvjv** [comment]
Question, @macdonst - Can you reproduce this with the steps I'd added? If not, what cordova version are you using?

**Comment by macdonst** [comment]
@jvjvjv I'm using cordova@7.0.1 & cordova-ios@4.4.0 and I'm not seeing the error. I'm not using Ionic though so I may have to try it with the Ionic CLI soonish. I'm at a conference right now so wifi is not great.

**Comment by jvjvjv** [comment]
I'm doing this without Ionic and still getting the error. Thanks for the assistance for now. I was hoping it would not take this long to troubleshoot.

**Comment by Sameendra** [comment]
I am also facing the same problem. For me it seems very spontaneous as sometimes it builds without complaining and sometimes it does not. It happens more often and could not find a reason. 

I am also using latest versions of cordova and cordova-Ios

**Comment by mfossel** [comment]
Same problem. Anyone with a solution or idea?

**Comment by xyboox** [comment]
Hi, it happens to me too, and strangely enough, not all the time ( but very often.. )

**Comment by fierysolid** [comment]
+1

**Comment by macdonst** [comment]
@shazron ever see anything like the above issue in cordova-ios?

**Comment by shazron** [comment]
Investigating, using steps:
```
cordova create bonsaitest com.bonsai.test "Boina"
cd bonsaitest
cordova plugin add phonegap-plugin-push@2.0.0-rc4
nano config.xml (add GoogleService-Info.plist file)
cordova build ios
```
Will update if I know more.

**Comment by shazron** [comment]
I was able to repro. Seems that others have this same problem: https://github.com/stetro/cordova-plugin-fcm-config/issues/3

**Comment by shazron** [comment]
When I tested it, it failed with an empty plist. 

When I copied the contents of my app's Info.plist into GoogleService-Info.plist, it builds fine (ran it ten times just in case). My hunch is the GoogleService-Info.plist you are using is malformed, thus the CopyPlistFile function errors out since it probably does some verification, not sure yet. 

Can you post an affected .plist that you used that exhibits this bug?

**Comment by shazron** [comment]
Scratch that, ran it > 10 times now I see the error. Hmm. I think I can see why it errors, perhaps a race condition since if I run it 10 times, I see 10 references in `Resources` in Xcode to the .plist and these multiple references probably are run concurrently resulting in "missing" files.

**Comment by shazron** [comment]
Filed https://issues.apache.org/jira/browse/CB-13112

**Comment by jvjvjv** [comment]
Super helpful, @shazron. Thank you!

I'm not sure of the internals of Cordova enough to sort of figure out a hackish way around the issue in the meantime?

**Comment by uj** [comment]
The phonegap build blog says to workaround this issue, use cli-65.0 with the new builder for phonegap-plugin-push.  Howerver, doing that causes the plugin to be skipped because the plugin requires cordova.

Plugin doesn't support this project's cordova-ios version. cordova-ios: 4.3.1, failed version requirement: >=4.4.0

So, the workaround on the phonegap blog really doesn't make sense.

**Comment by macdonst** [comment]
@uj no the workaround does not make sense. I've asked the author to clear up that misconception.

**Comment by shazron** [comment]
Identified the problem: https://issues.apache.org/jira/browse/CB-13112?focusedCommentId=16117720&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16117720

TLDR; unit tests used an older project template which had a flaw that didn't surface this problem.

**Comment by shazron** [comment]
Fix: https://github.com/apache/cordova-ios/pull/329 (all tests pass)

This will be in the next cordova-ios release. I can put out a hotfix (cordova-ios@4.4.0 that includes this fix -- I will just branch off the 4.4.0 tag and cherry-pick this fix, and you can install off the hotfix branch for the platform. non-official release, for testing only).

For now, you can install this platform to test the fix:
```
cordova platform add https://github.com/shazron/cordova-ios.git\#CB-13112
```

Please test, thanks!

**Comment by tlaverdure** [comment]
What is the status if this?

**Comment by uj** [comment]
@tlaverdure They fixed this on Aug-8 in PhoneGap Build, and there was also a patch at the same time for people using their own build computer.

**Comment by tlaverdure** [comment]
Well for whatever it is worth is seems like building ionic and Cordova separately solved this for me.

`ionic cordova build --prod --release --device` was failing when the GoogleService-Info.plist file couldn't be copied. However, `ionic build --prod && cordova build ios --release --device` seems to work. 😕

**Comment by uj** [comment]
@tlaverdure Are you using the CB-13112 patch?  Or are you using the official release of cordova?

**Comment by tlaverdure** [comment]
The official release, which has been buggy for some time now.

**Comment by uj** [comment]
@tlaverdure Yes, that is why you have to install the cb-13112 patch to fix it, then it will work properly.

**Comment by JaneDawson** [comment]
Working great, @shazron. Thank you!!

**Comment by shyamal890** [comment]
Is this solved with the new release of cordova-ios 4.5.0 ?

**Comment by shazron** [comment]
@shyamal890 yes.

**Comment by AmitAKB** [comment]
Error: Source path does not exist: GoogleService-Info.plist
Im getting above error when i am running cordova build android command

**Comment by jvjvjv** [comment]
@AmitAKB This ticket has been closed already. If the information here does not help you please open a new ticket.
